File: oop/Ad.java
package oop;


public class Ad extends BaseAdvertising{

    private String title; 
    private String imgUrl; 
    private String link; 
    private Advertiser advertiser;


    public Ad(int ID, String title, String imgUrl, String link, Advertiser advertiser) {
        super.setID(ID);
        this.title = title;
        this.imgUrl = imgUrl;
        this.link = link;
        this.advertiser = advertiser;
    }


    public String getTitle() {
        return this.title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getImgUrl() {
        return this.imgUrl;
    }

    public void setImgUrl(String imgUrl) {
        this.imgUrl = imgUrl;
    }

    public String getLink() {
        return this.link;
    }

    public void setLink(String link) {
        this.link = link;
    }   


    public Advertiser getAdvertiser() {
        return this.advertiser;
    }

    public void setAdvertiser(Advertiser advertiser) {
        this.advertiser = advertiser;
    }

    @Override 
    public void incClicks() {
        super.incClicks();
        this.advertiser.incClicks();
    }

    @Override
    public void incViews() {
        super.incViews();
        this.advertiser.incViews();
    }

    @Override
    public String describeMe() {

        return "Ad Class Represents The Entity of Ad";
    }

}

File: python/ad.py
#!/usr/bin/python3 
from baseadvertising import BaseAdvertising

class AD(BaseAdvertising):

    def __init__(self, id, title, imgUrl, link, advertiser):
        super().__init__(id)
        self._title = title
        self._imgUrl = imgUrl
        self._link = link
        self._advertiser = advertiser

    def gettTitle(self):
        return self._title 

    def setTitle(self, title):
        self._title = title

    def getImgUrl(self):
        return self._imgUrl

    def setImgUrl(self, imgUrl):
        self._imgUrl = imgUrl

    def getlink(self):
        return self._link 

    def setlink(self, link):
        self._link = link

    def getAdvertiser(self):
        return self._advertiser 

    def setAdvertiser(self, adv):
        self._advertiser = adv     
    
    def incClicks(self):
        self._advertiser.incClicks()
        super().incClicks()

    def incViews(self):
        self._advertiser.incViews() 
        super().incViews()

    @staticmethod
    def describeMe():
        return "Ad Class Represents The Entity of Ad"
